### DISCUSSION TITLE
val_dataloader` has `shuffle=True` though its false

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Lightning-AI/pytorch-lightning#11392] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am working with [pytorchvideo](https://github.com/facebookresearch/pytorchvideo) with pytorch lightning but it say ```UserWarning: Your `val_dataloader` has `shuffle=True`,it is strongly recommended that you turn this off for val/test/predict dataloaders.```
```
from pytorchvideo.models.resnet import create_resnet
class OurModel(LightningModule):
    def __init__(self):
        super(OurModel,self).__init__()
        self.model =  create_resnet(
                      input_channel=3, # RGB input from Kinetics
                      model_depth=50, # For the tutorial let's just use a 50 layer network
                      model_num_class=1, # Kinetics has 400 classes so we need out final head to align
                      norm=nn.BatchNorm3d,
                      activation=nn.ReLU)

    def forward(self,x):
        return self.model(x)

  
    def val_dataloader(self):
        val_dataset=LabeledVideoDataset(labeled_video_paths=\
                    list(zip(val_df.vidpath,val_df.pulse)),
                   clip_sampler=make_clip_sampler("uniform", 2),\
                    transform=train_transform,  decode_audio=False)
        
        val_loader=DataLoader(val_dataset,shuffle=False,
                   batch_size=self.batch_size,
                   num_workers=self.numworker,
                   pin_memory=True)
        return val_loader
    
    def validation_step(self,batch,batch_idx):
        out = self(batch["video"]).flatten()
        loss=self.criterion(out,batch["label"].float())
        mae=self.metric(out,batch["label"].float())
        return {'loss':loss,'mae':mae.detach()}

```
A you can see, shuffle is False. but it keep me giving warning that
```
UserWarning: Your `val_dataloader` has `shuffle=True`,it is strongly recommended that you turn this off for val/test/predict dataloaders.
```
Sorry, i am not sure whether i had to ask it at pytorchvideo or here

**Comment by akihironitta** [comment, marked as answer]
Hi @talhaanwarch, you're asking it in the right place!

It's a warning from Lightning, and as I looked at the definition of `make_clip_sampler` of pytorchvideo, I believe it's the same reason as https://github.com/PyTorchLightning/pytorch-lightning/discussions/10771. You can simply ignore it with some filter like below if you need.
```python
import warnings
warnings.simplefilter('ignore', category=UserWarning, message="Your `val_dataloader` has `shuffle=True`.*")
```
